### ISSUE TITLE
Possible flaky MachineHealthCheck Test

### ISSUE THREAD

**Issue by wfernandes** [kubernetes-sigs/cluster-api#4028]
**What steps did you take and what happened:**
As I was reviewing PRs, I came across some test failures where I had to kick off the test-main job after which it passed.

See the prow job below.
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/3752/pull-cluster-api-test-main/1339633406769631232

**What did you expect to happen:**
Expected it to pass.

**Anything else you would like to add:**
```
    machinehealthcheck_controller_test.go:219: 
        Timed out after 5.001s.
        expected
        	&v1alpha4.MachineHealthCheckStatus{ExpectedMachines:2, CurrentHealthy:2, RemediationsAllowed:2, ObservedGeneration:1, Targets:[]string{"test-mhc-machine-tskbr", "test-mhc-machine-wzbmh"}, Conditions:v1alpha4.Conditions{v1alpha4.Condition{Type:"RemediationAllowed", Status:"True", Severity:"", LastTransitionTime:v1.Time{Time:time.Time{wall:0x0, ext:63743825408, loc:(*time.Location)(0x2832fe0)}}, Reason:"", Message:""}}}
        to match
        	&v1alpha4.MachineHealthCheckStatus{ExpectedMachines:2, CurrentHealthy:2, RemediationsAllowed:2, ObservedGeneration:1, Targets:[]string{"test-mhc-machine-wzbmh", "test-mhc-machine-tskbr"}, Conditions:v1alpha4.Conditions{v1alpha4.Condition{Type:"RemediationAllowed", Status:"True", Severity:"", LastTransitionTime:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, Reason:"", Message:""}}}
```
After a quick glance as to why it failed, I noticed that the MachineHealthCheckStatus was the same (for the most part). 
The only thing I noticed was that the `Targets` were ordered differently and the `LastTransitionTime` is different. However, in the code we are using a `MatchMachineHealthCheckStatus` gomega matcher which ignore the `LastTransitionTime`.

I'm wondering if we should `ConsistOf` matcher for the `Targets`. 🤔 

/kind bug
/help
/good-first-issue
/area testing

**Comment by fabriziopandini** [comment]
@wfernandes we recently merged https://github.com/kubernetes-sigs/cluster-api/pull/3999 (backport https://github.com/kubernetes-sigs/cluster-api/pull/3998) for this very same problem, but +1 to check this again if you find new eveidences

**Comment by wfernandes** [comment]
@fabriziopandini Yeah...it definitely seems like #3999 should've fixed it and it was merged 8 days ago! The prow job failure happened a day ago so I'm not sure what's up with that. I'll try running the test locally to see if it still pops up and if I can't get it to reproduce then I'll close this issue.

**Comment by wfernandes** [comment]
I think I figured it out. 

I reran the test for this PR #3752 and got the same result again.
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/3752/pull-cluster-api-test-main/1340044962326122496

Then I realized that the PR is editing `machinehealthcheck_controller.go` and `machinehealthcheck_controller_test.go` so I'm guessing the rebase with the main branch is blowing away the `sort.Strings` changes. 

I'll close this issue since the examples of the failure are associated with that PR specifically.
/close
